### DISCUSSION TITLE
The dream: pulumi import -all

### DISCUSSION THREAD

**Discussion by davidkhala** [pulumi/pulumi#12058] [Ideas]
In some cloud providers, there is a way to view all resources in an account
- In Azure, there is an all resources view for subscription.
- In Oracle Cloud, by compartment we can have tenancy explorer.

So I wonder, after we have configured the credential and provider select, can we have a 
`pulumi import --all` options that discover all resource that is visible to current credential?

It can bring huge benefits compared to import resources one by one.

**Comment by hermanbanken** [comment]
If `pulumi import --all` would import just the resources that are defined in the code, that would already be of great help. The syntax `pulumi import <type> <name> <name?>` is annoying if you just want to import everything that exists in the code and you're lazy and do not want to manually copy paste all those import commands.

**Reply by Frassle** [reply]
I'm currently working on `pulumi preview --import-file import.json` which looks at every resource that needs to be created and writes out a placeholder import file with all the names and types that just needs the IDs filling in to then import everything.
